pyannote/audio/embedding/approaches/contrastive_loss.py

```diff
@@ -122,24 +105,29 @@ def batch_loss(self, batch):
             ['loss'] (`torch.Tensor`) : Triplet loss
         """
 
-        fX = self.forward(batch)
-        y = batch['y']
+        fX, y = self.embed(batch)
 
         # calculate the distances between every sample in the batch
-        nbatch = fX.size(0)
+        batch_size = fX.size(0)
         dist = self.pdist(fX).to(self.device_)
 
         # calculate the ground truth for each pair
+        # TODO. this can be done much more cleanly with
+        # pyannote.core.utils.distance.pdist(y, metric='equal')
         gt = []
-        for i in range(nbatch - 1):
-            for j in range(i + 1, nbatch):
+        for i in range(batch_size - 1):
+            for j in range(i + 1, batch_size):
                 gt.append(int(y[i] != y[j]))
         gt = torch.Tensor(gt).float().to(self.device_)
 
         # Calculate the losses as described in the paper
         losses = (1 - gt) * torch.pow(dist, 2) + gt * torch.pow(torch.clamp(self.margin_ - dist, min=1e-8), 2)
+
+        # FIXME: why divive by 2?
```

Comment:
The division by 2 comes from equation 4 in the [original paper](http://yann.lecun.com/exdb/publis/pdf/hadsell-chopra-lecun-06.pdf)